peer/peerdiscovery.go

```diff
@@ -0,0 +1,147 @@
+package peer
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"github.com/renproject/surge"
+	"time"
+
+	"github.com/renproject/aw/transport"
+	"github.com/renproject/aw/wire"
+	"github.com/renproject/id"
+
+	"go.uber.org/zap"
+)
+
+type DiscoveryClient struct {
+	opts DiscoveryOptions
+
+	transport *transport.Transport
+}
+
+func NewDiscoveryClient(opts DiscoveryOptions, transport *transport.Transport) *DiscoveryClient {
+	return &DiscoveryClient{
+		opts: opts,
+		transport: transport,
+	}
+}
+
+func (dc *DiscoveryClient) DiscoverPeers(ctx context.Context) {
+	var pingData [2]byte
+	binary.LittleEndian.PutUint16(pingData[:], dc.transport.Port())
+
+	msg := wire.Msg{
+		Version: wire.MsgVersion1,
+		Type:    wire.MsgTypePing,
+		Data:    pingData[:],
+	}
+
+	ticker := time.NewTicker(dc.opts.PingTimePeriod)
+	defer ticker.Stop()
+
+	alpha := dc.opts.Alpha
+	sendDuration := dc.opts.PingTimePeriod/time.Duration(alpha)
+	outer:
```

Comment:
Nice use of outer loop labelling. This is an interesting way to restart the ping process if it exceeds the tick interval. Idiomatically, this label should be capitalised.